roles/kubernetes/preinstall/tasks/dhclient-hooks-undo.yml

```diff
@@ -3,19 +3,24 @@
 # These tasks will undo changes done by kargo in the past if needed (e.g. when upgrading from kargo 2.0.x
 # or when changing resolvconf_mode)
 
+- name: Check if dhclient conf file exists
+  stat: path={{dhclientconffile}}
+  register: dhclientconf_stat
+
 - name: Remove kargo specific config from dhclient config
   blockinfile:
     dest: "{{dhclientconffile}}"
     state: absent
     backup: yes
     follow: yes
     marker: "# Ansible entries {mark}"
+  when: dhclientconf_stat.stat.exists
```

Comment:
On some systems (@mnort which cloud/distro did you see this issue on?), this file does not exist before kargo initially creates it. In this case, the current kargo version only creates this file with resolvconf_mode=host_resolvconf, so it will never be there on fresh installed nodes and thus this task fails.

Which name do you prefer? I don't like to simply use `dhclientconf` as I'm always afraid of ambiguity. What about dhclientconf_test?